appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/EJBContainerTransactionManager.java

```diff
@@ -110,63 +100,63 @@
      * tx attributes can be looked up with variations of getTxAttr()
      */
     int findTxAttr(MethodDescriptor md) {
-        int txAttr = -1;
-
-        if ( container.isBeanManagedTran ) {
+        if (container.isBeanManagedTran) {
             return Container.TX_BEAN_MANAGED;
         }
-
-        ContainerTransaction ct = ejbDescriptor.getContainerTransactionFor(md);
-
-        if ( ct != null ) {
-            String attr = ct.getTransactionAttribute();
-            if ( attr.equals(ContainerTransaction.NOT_SUPPORTED) )
-                txAttr = Container.TX_NOT_SUPPORTED;
-            else if ( attr.equals(ContainerTransaction.SUPPORTS) )
-                txAttr = Container.TX_SUPPORTS;
-            else if ( attr.equals(ContainerTransaction.REQUIRED) )
-                txAttr = Container.TX_REQUIRED;
-            else if ( attr.equals(ContainerTransaction.REQUIRES_NEW) )
-                txAttr = Container.TX_REQUIRES_NEW;
-            else if ( attr.equals(ContainerTransaction.MANDATORY) )
-                txAttr = Container.TX_MANDATORY;
-            else if ( attr.equals(ContainerTransaction.NEVER) )
-                txAttr = Container.TX_NEVER;
+        final int txAttr = getTransactionAttribute(md);
+        if (txAttr < 0) {
+            throw new EJBException("Transaction Attribute not found for method " + md.prettyPrint());
         }
-
-        if ( txAttr == -1 ) {
-            throw new EJBException("Transaction Attribute not found for method "
-                + md.prettyPrint());
-        }
-
         container.validateTxAttr(md, txAttr);
-
         return txAttr;
     }
 
+    private int getTransactionAttribute(final MethodDescriptor md) {
```

Comment:
This split allowed to use a single assignment to a final. It is also much more comprehensible - one method only looks for a TransactionAttribute value and the caller resolves how to use it.